llvm/test/tools/llvm-split/SYCL/device-code-split/module-split-func-ptr.ll

```diff
@@ -0,0 +1,43 @@
+; This test checks that Module splitting can properly perform device code split by tracking
+; all uses of functions (not only direct calls).
+
+; RUN: llvm-split -sycl-split=source -S < %s -o %t
+; RUN: FileCheck %s -input-file=%t_0.sym --check-prefix=CHECK-SYM0
+; RUN: FileCheck %s -input-file=%t_1.sym --check-prefix=CHECK-SYM1
+; RUN: FileCheck %s -input-file=%t_0.ll --check-prefix=CHECK-IR0
+; RUN: FileCheck %s -input-file=%t_1.ll --check-prefix=CHECK-IR1
+
+; CHECK-SYM0: kernel2
+; CHECK-SYM1: kernel1
+;
+; CHECK-IR0: define dso_local spir_kernel void @kernel2
```

Comment:
It makes sense to rename kernel2 to kernel0 to align with the check prefixes (or use 2 in the prefix name instead of 0).

Reply:
Now I use letter 'A', 'B' for kernels.